zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Identity.java

```diff
@@ -98,26 +97,42 @@ private void findUser(HttpServletRequest request, HttpServletResponse response)
     }
 
     private ServiceAccount findService(HttpServletRequest request) throws IOException {
-        final ServiceStore serviceStore = getServiceStoreImpl();
         String authorizationHeader = request.getHeader(AUTHORIZATION_HEADER);
-        if (StringUtils.isNotEmpty(authorizationHeader)) {
-            authorizationHeader = authorizationHeader.toLowerCase();
-            if (authorizationHeader.toLowerCase().startsWith("bearer ")) {
-                String[] cred = authorizationHeader.split("bearer ");
-                if (cred.length != 2) {
-                    return null;
-                }
-                String token = cred[1];
-                final ServiceAccount service = serviceStore.get(token);
-                if (service != null) {
-                    info().data("user", service.getID()).log("authenticated service account confirmed");
-                    return service;
-                } else {
-                    return null;
-                }
-            }
+        if (StringUtils.isEmpty(authorizationHeader)) {
+            warn().log("service auth header is empty returning null");
+            return null;
         }
-        return null;
+
+        authorizationHeader = authorizationHeader.toLowerCase();
+        if (!authorizationHeader.startsWith("bearer ")) {
+            warn().log("authorization header does not begin with the required bearer prefix returning null");
+            return null;
+        }
+
+        String serviceToken = removeBearerPrefix(authorizationHeader);
+        ServiceAccount serviceAccount = null;
+
+        try {
+            serviceAccount = getServiceStore().get(serviceToken);
```

Comment:
On line 106 you are lowercasing the auth header therefore the service token has now been lowercased.  However from looking at the service store and service endpoint it does not look like we are enforcing that the token be lowercased there.  Since this is a direct lookup we probably want to be explicit about lowercasing everywhere.  Potentially it would be best to add this to the ServiceStoreImpl.

Reply:
Existing logic. I haven't change it just rearranged it for readability.

Reply:
The lower casing is actually wrong and not needed. I can see why it was done originally - to make it easier to check for both `"Bearer"` and `"bearer"` however thats not correct. According to the specification the correct and expected value is `Bearer XXX`.